core/state_transition.go

```diff
@@ -327,7 +327,7 @@ func (st *StateTransition) preCheck() error {
 	if st.evm.ChainConfig().IsCancun(st.evm.Context.BlockNumber, st.evm.Context.Time) {
 		if st.blobGasUsed() > 0 {
 			// Skip the checks if gas fields are zero and blobBaseFee was explicitly disabled (eth_call)
-			if !st.evm.Config.NoBaseFee && msg.BlobGasFeeCap.BitLen() > 0 {
+			if !st.evm.Config.NoBaseFee || msg.BlobGasFeeCap.BitLen() > 0 {
```

Comment:
I had to, again, look at it for a while before my brain parsed it out fully. Yeah it's ok now, but I'd prefer having it rewritten for clarity:  
```
// skip checks if basefee was disabled and blobBaseFee is zero (eth_call) 
skipCheck := st.evm.Config.NoBaseFee && msg.BlobGasFeeCap.BitLen() == 0
if !skipCheck {
```

Reply:
Fair enough

Reply:
Pushed this fix and also added it to the 1559 basefee